### ISSUE TITLE
Failed resolution of global activate should indicate if package is discontinued

### ISSUE THREAD

**Issue by parlough** [dart-lang/pub#4130]
There's been a lot of confusion and reported issues by developers finding old references to `dart pub global activate` for [`package:devtoools`](https://pub.dev/packages/devtools) which was discontinued in favor of being bundled with the SDK.

A recent example of this resulting in confusion is https://github.com/flutter/devtools/issues/6550.

On recent SDK versions this fails due to incompatible SDK constraints, potentially misleading the user into thinking they have an issue. The error message should likely also indicate a package is discontinued and if it has a replacement. 

I hope that will help avoid some of the confusion.

There's a separate issue for devtools though (likely for pub.dev), as there is no replacement package, so it might be nice if instead of a replacement, a link for more information could be configured instead.

**Comment by sigurdm** [comment]
Yeah it would be nice to show up-front that the package is discontinued.

A sad thing here is that **if** it managed to resolve it would be shown as discontinued in the resolution report.

> There's a separate issue for devtools though (likely for pub.dev), as there is no replacement package, so it might be nice if instead of a replacement, a link for more information could be configured instead. This might also help for packages and APIs that start outside of an SDK and later are included, if that ever happens.

I think we decided against a custom deprecation message because we don't want package authors to be able to put arbitrary stuff in the terminal output. (could eg. be used for advertisement). Maybe we could link to the package page - where the README could explain more...

**Comment by sigurdm** [comment]
Related to https://github.com/dart-lang/pub-dev/issues/2457
